pkgs/by-name/ra/raccoin/package.nix

```
@@ -0,0 +1,103 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  makeWrapper,
+  libxkbcommon,
+  wayland,
+  cairo,
+  pango,
+  harfbuzz,
+  fontconfig,
+  freetype,
+  alsa-lib,
+  libGL,
+  xorg,
+  openssl,
+  withWayland ? true,
+  withX11 ? true
+}:
+
+let
+  # dependencies, grouped
+  coreDeps = [
+    cairo
+    pango
+    harfbuzz
+    fontconfig
+    freetype
+    alsa-lib
+    libGL
+    
+    openssl
+    libxkbcommon
+  ];
+
+  waylandDeps = lib.optionals withWayland [
+    wayland
+  ];
+
+  x11Deps = lib.optionals withX11 [
+    xorg.libX11
+    xorg.libXcursor
+    xorg.libXi
+    xorg.libXrandr
+  ];
+
+  allDeps =
+    coreDeps ++
+    waylandDeps ++
+    x11Deps
+  ;
+in
+#stdenv.mkDerivation rec {
+rustPlatform.buildRustPackage {
+  pname = "raccoin";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "bjorn";
+    repo = "raccoin";
+    rev = "v0.2.0";
+    hash = "sha256-6BwRFU8qU6K0KqKdK+soKcWU2LPxkKKPOcn2gupunGg=";
+  };
+  cargoHash = "sha256-pz3dwIIBQZIwTammiI/UQwM0Iy1ZgC9ntK+qNGv3s24=";
+
+  nativeBuildInputs = [
+    pkg-config
+    makeWrapper
+  ];
+  buildInputs = allDeps;
+
+  # deactivated tests by cargo as build sandbox doesn not have the necessarry display
+  # also no tests available yet, so checks may result in failing due to missing tests
+  doCheck = false;
+  # alternatively in case tests are added later on
+  #doCheck = true;
+  #checkPhase = ''
+  #  cargo test --lib -- --nocapture
+  #'';
+
+  postFixup = ''
+    wrapProgram $out/bin/raccoin \
+      --set-default SLINT_BACKEND winit \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath allDeps}
+  '';
+
+  # not sure how to apply, but this may make the shell script for updates obsolete
+  #passthru.updateScript = nix-update-script { };
```

this is literally how it's applied
```suggestion
  passthru.updateScript = nix-update-script { };
```